[CS2113T-W11-1] SysLib CLI

yingx9 opened the pull request:
SysLib CLI is a program designed specially for system librarians to manage their work and
responsibilities. Using intuitive commands, view, add, delete, and find books from the library
inventory without any hassle.

Musfirahe0556596 commented on src/main/java/seedu/parser/Parser.java:
```diff
@@ -0,0 +1,155 @@
+
+public class Parser {
+
+    public List<Resource> resourceList = new ArrayList<>();
```
Plural form should be used on names representing a collection of objects.

Musfirahe0556596 commented on src/main/java/seedu/parser/Parser.java:
```diff
@@ -0,0 +1,155 @@
+        boolean matchFound = matcher.find();
+
+        if (matchFound) {
+            if (matcher.group(2).equalsIgnoreCase("b")) {
```
"b" is a magic literal - would be good if its placed as a named constant

Musfirahe0556596 commented on src/main/java/seedu/parser/Parser.java:
```diff
@@ -0,0 +1,155 @@
+            boolean matchFound = matcher.find();
+
+            Pattern gPattern = Pattern.compile(genrePattern);
+            Matcher gMatcher = gPattern.matcher(matcher.group(5));
+            boolean gMatchFound = gMatcher.find();
+
+            String[] args = new String[6];
+
+            if (matchFound) {
+                args[0] = matcher.group(1).trim(); // id
+                args[1] = matcher.group(2).trim(); // title
+                args[2] = matcher.group(3).trim(); // author
+                args[3] = matcher.group(4).trim(); // tag
+                if (gMatchFound) {
+                    args[4] = gMatcher.group(1).trim(); // isbn
+                    args[5] = gMatcher.group(2).trim(); // genre
+                } else {
+                    args[4] = matcher.group(5).trim(); // isbn
```
Magic literals (e.g. group(<5>)) - would be good if the numbers are placed as a named constant.

Musfirahe0556596 commented on src/main/java/seedu/parser/Parser.java:
```diff
@@ -0,0 +1,155 @@
+            put("add", new AddCommand());
+        }
+    };
+    public void process(String response) {
```
Could rename the method process() better (e.g. process what?)

Musfirahe0556596 commented on src/main/java/seedu/commands/Command.java:
```diff
@@ -0,0 +1,91 @@
+    }
+    public int parseInt(String value){
+        try {
+            int num = Integer.parseInt(value);
```
Could rename the variable num better to indicate what it is

Musfirahe0556596 commented on src/main/java/seedu/commands/ListCommand.java:
```diff
@@ -0,0 +1,133 @@
+    @Override
+    public void execute(String statement, Parser parser) throws SysLibException, IllegalArgumentException {
+        feedbackToUser = "";
+        String[] values = parseArgument(statement);
```
Could rename the variable "values" better - what kind of values?.

Musfirahe0556596 commented on src/main/java/seedu/commands/FindCommand.java:
```diff
@@ -0,0 +1,120 @@
+    public void execute(String statement, Parser parser) throws IllegalArgumentException, SysLibException {
+        assert statement != null && !statement.trim().isEmpty() : "Statement to execute cannot be null or empty!";
+        assert parser != null : "Parser cannot be null!";
+        String[] value = parseArgument(statement);
```
Plural form should be used on names representing a collection of objects.

Musfirahe0556596 commented on src/main/java/seedu/commands/FindCommand.java:
```diff
@@ -0,0 +1,120 @@
+        String[] value = parseArgument(statement);
+        validate(statement, value);
+
+        if (value[3]==null && value[2]==null && value[1]==null && value[0]==null) {
```
Magic literals e.g. value[<3>]- would be good if it is a named constant.

Musfirahe0556596 commented on src/main/java/seedu/commands/FindCommand.java:
```diff
@@ -0,0 +1,120 @@
+        ArrayList<Resource> matchedResources = new ArrayList<>();
+        for (Resource r: parser.resourceList){
+            Book b = (Book) r;
+            if (b.getTitle().equals(value[3]) || b.getISBN().equals(value[1]) || b.getAuthor().equals(value[2])){
```
Magic literals e.g. value[<3>]- would be good if it is a named constant.

Musfirahe0556596 commented on src/main/java/seedu/commands/Command.java:
```diff
@@ -0,0 +1,91 @@
+    /*
+    validate will include str
+     */
+    public void validate(String statement, String[] value) throws IllegalArgumentException{
```
Could rename the method validate() better (e.g. validate what?)

Musfirahe0556596 replied:
The argument String[] value should be in plural form.

Musfirahe0556596 commented on src/main/java/seedu/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,38 @@
+    public void execute(String statement, Parser parser) throws SysLibException {
+        int id = parseInt(parseArgument(statement)[0]);
+        assert id > 0;
+        ArrayList<Resource> toRemove = new ArrayList<>();
```
Plural form should be used on names representing a collection of objects.

Musfirahe0556596 reviewed (commented):
Good! But here are some code quality violations to consider (I have also created Issues for each individual member)

mcmc101001 commented on docs/images/AddSequenceDiagram.png:
would it be better for the arrow in the condition where containsKey is false to show some kind of exception being invoked?

mcmc101001 commented on docs/images/AddSequenceDiagram.png:
Would it be better to have some textual explaination of notEqualsIgnoreCase in the developer guide as it is quite ambiguous from the diagram what purpose it serves

mcmc101001 commented on docs/images/ListSequenceDiagram.png:
Should the containsKey == false section be below the portion with the Command portion, because the Command portion only comes into play in the case where containsKey is true. Currently the containsKey == false portion is overlapping with the Command block

mcmc101001 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,232 @@
+
+`add` has seven options:
+- add /id [id] /t [title] /a [author] /tag [tag] /i [isbn]
+- add /id [id] /t [title] /a [author] /tag [tag] /i [isbn] _/g [genre] /s [status]_
```
What does the italics represent? Perhaps more explaination can be added to the DG to explain it. I am not sure if it is there to highlight the difference or to indicate that they are optional tags.

skylee03 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
-## Design & implementation
+## Setting Up & Getting Started
+
+1. Fork the repo at https://github.com/AY2324S1-CS2113T-W11-1/tp.
```
Would it be better to use hyperlinks?

skylee03 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+
+SysLib currently consists of four main components:
+
+- `UI`: User Interaction
+- `Parser`: Parsing User Response 
+- `Command`: Command Executor
+- `Data`: Holds the data of SysLib
```
Would it be better to indicate where other packages (e.g., `storage`, `common`, etc.) belong?

skylee03 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+The `find` command allows users to search for resources based on specified filters such as author (`/a`), ISBN (`/i`), ID (`/id`), and title (`/t`). The results will show all resources that match any of the given filters.
+
+`find` has the following options:
+- `find /id [ID]`
+- `find /t [TITLE]`
+- `find /a [AUTHOR]`
+- `find /i [ISBN]`
```
According to the [textbook](https://se-education.org/guides/conventions/markdown.html), we are suggested to use `*` instead of `-` for bullet-points, but it is optional to follow it.

KenCheung18 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+
+SysLib currently consists of four main components:
+
+- `UI`: User Interaction
```
UI should be User Interface?

skylee03 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+
+Upon receiving the `find` command, the system will:
+
+1. Parse the filters and their associated values.
+2. Filter the resources based on the given filters.
+3. Display the matching resources.
```
According to the textbook, we are suggested to use generic numbering for ordered lists, but it is optional to follow it.

skylee03 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+
+#### Example Usage Scenario
+
+Step 1. The user inputs the command: `add /id 0005 /t Frankenstein /a Mary Shelley /i FKS0001 /tag B /g Gothic, Fiction`
```
Why is "Step 1." in bold above but not here? Is it better to have a unified style?

skylee03 commented on docs/images/AddSequenceDiagram.png:
This picture is too wide! Try to avoid repeating the parameters that are too long.

skylee03 replied:
One activation bar corresponds to one method call. Therefore `:Command` should have 2 activation bars.

skylee03 replied:
`AddCommand` is a subclass of `Command`, and the `:AddCommand` instance is just calling the methods of it self. Would it be more accurate to regard `parseArgument` and `validateStatement` as its members?

skylee03 replied:
Would it be more accurate to explicitly mark `:AddCommand` as _destroyed_?

KenCheung18 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+
+
+   
+### UI Component
+
+### Parser Component
+
+### Command Component
+
+
+### Data Component
```
Maybe can add some content after each title

skylee03 replied:
> Maybe can add some content after each title

I think they just have not started to write them… 😂

TongZhengHong commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+- `Command`: Command Executor
+- `Data`: Holds the data of SysLib
+
+
```
Consider adding class diagrams to show how the different classes are related to each other.

skylee03 commented on docs/images/FindSequenceDiagram.png:
One activation bar corresponds to one method call. Therefore `:Command` should have 2 activation bars.

skylee03 replied:
Would it be more accurate to use `->` instead of `-->` for the arrow of `warning`?

skylee03 replied:
`FindCommand` is a subclass of `Command`, and the `:FindCommand` instance is just calling the methods of it self. Would it be more accurate to regard `parseArgument` and `validateStatement` as its members?

yicheng-toh commented on docs/images/ListSequenceDiagram.png:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/75836313/8b71e3eb-f869-4c04-802c-e09c0cb56fcb)
One of the bar has a return arrow while another doesn't. Perhaps you could add the arrow in the second circle or you could add return type in the first circle.

TongZhengHong commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+#### Sequence Diagram
+The following sequence diagram shows how the add function works:
+<img src="images/AddSequenceDiagram.png"/>
+
```
This diagram contains a repeated input string which is very long. Can consider mentioning the input outside of the diagram to make the diagram more compact and readable

skylee03 commented on docs/images/ListSequenceDiagram.png:
One activation bar corresponds to one method call. Therefore `:Command` should have 2 activation bars.

skylee03 replied:
Would it be more accurate to complete the return arrows?
![1698917627170](https://github.com/nus-cs2113-AY2324S1/tp/assets/24489025/14d325a2-aee8-431b-9cd2-1e8fe3edd20d)

yicheng-toh commented on docs/images/ListSequenceDiagram.png:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/75836313/f91603fb-9353-476e-af42-2439173d05d7)
Should the else box be extended all the way?

KenCheung18 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+
+#### Sequence Diagram
+
+<img src="images/FindSequenceDiagram.png" />
```
Should add a conditional block for returning legalArgumentException? Since it's try catch block so not necessary to throw exception

antrikshdhand replied:
This sequence diagram is quite verbose. All return values and method calls are written with sample arguments provided, which is not necessary for a high-level diagram.

KenCheung18 replied:
if you are calling a function resourceList should be like resourceList() in the sequence diagram

skylee03 commented on docs/AboutUs.md:
The table is not displayed properly. Please try to add an empty line between the table and the `# About us`.

skylee03 replied:
The avatars are too big! Maybe making the picture smaller will look better!

yicheng-toh reviewed (commented):
Good job on the DG.

TongZhengHong commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+<img src="images/FindSequenceDiagram.png" />
+
+### Examples for Testing
+
```
Good use of examples!

skylee03 commented on docs/UserGuide.md:
````diff
@@ -1,42 +1,302 @@
+5. Run the following command: `java -jar Syslib.jar`. You should see the following welcome screen.
+```
+____________________________________________________________
+ ____            _     _ _        ____ _     ___ 
+/ ___| _   _ ___| |   (_) |__    / ___| |   |_ _|
+\___ \| | | / __| |   | | '_ \  | |   | |    | | 
+ ___) | |_| \__ \ |___| | |_) | | |___| |___ | | 
+|____/ \__, |___/_____|_|_.__/   \____|_____|___|
+       |___/                                     
````
Spectacular ASCII art!

TongZhengHong commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+#### Implementation
+
+It implements the following operations:
+
```
Is `ADDCOMMAND` needed for each operation? Might want to explicitly mention that each operation belongs to `ADDCOMMAND` once at the start instead

antrikshdhand commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+
+#### Sequence Diagram
+The following sequence diagram shows how the add function works:
+<img src="images/AddSequenceDiagram.png"/>
```
It's a good diagram in many ways, however the size could be much more compact without low-level details such as exactly what method is called or what text is passed into each argument.

TongZhengHong commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+
+`add` has seven options:
+- add /id [id] /t [title] /a [author] /tag [tag] /i [isbn]
+- add /id [id] /t [title] /a [author] /tag [tag] /i [isbn] _/g [genre] /s [status]_
```
Can consider putting these commands in a code block for readability.

antrikshdhand commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,231 @@
+
+#### Example Usage Scenario
+
+Step 1. The user inputs the command: `add /id 0005 /t Frankenstein /a Mary Shelley /i FKS0001 /tag B /g Gothic, Fiction`
+
+Step 2. The `UI` component forwards the input to `SYSLIB`, which in turn passes it to the `PARSER`.
+
+Step 3. The `PARSER` processes the command and determines that it contains a valid key (`add`). It then calls 
+`ADDCOMMAND#execute(statement: String, this: Parser)` with the input command.
+
+Step 4. The `ADDCOMMMAND` component receives the command and performs the following operations:
+- Calls `ADDCOMMAND#parseArgument(statement: String)` to extract values for ID, title, author, ISBN, tag, and genres.
+- Calls `ADDCOMMAND#validate(statement: String, values: String[])` to ensure the validity of the input command.
+
+Step 5. The `COMMAND` component processes the input command to ensure that it meets the required format and constraints.
+It prepares the argument values for further processing.
+
+Step 6. Since the `tag` argument in the input command indicates that it is a book, the `ADDCOMMAND` determines that the
+key is equal to `b` (ignoring case). It then creates a new `Book` object using the parsed values (title, ISBN, author, 
+genres, ID).
+
+Step 7. The newly created book is forwarded to the `PARSER` to be added to the `resourceList`.
```
A well-written, detailed explanation of the process. Great for a future developer!

antrikshdhand reviewed (commented):
Overall a nice DG, however the verbose sequence diagrams could be improved.

skylee03 reviewed (commented):
LGTM!

KenCheung18 commented on docs/DeveloperGuide.md:
````diff
@@ -33,6 +238,104 @@
+```
+Example response:
+```
+Commands available:
+add: adds a new resource to the library inventory.(e.g. add /id ID /t TITLE /a AUTHOR /tag TAG /i ISBN [/g GENRE])
+delete: deletes the resource with the specified ID from the library inventory. (e.g. delete /id 123456789)
+list: list all resources OR filter by certain tags or genre.(e.g. list /tag B /g Fiction
+find: find a resource by title, author, ISBN or given id. (e.g. find /i 9780763630188)
+edit: Edit a listing by entering its isbn to update its details. (e.g. edit /i 123 /t NEW_TITLE /a NEW_AUTHOR)
+exit: displays a farewell message and exits the program (e.g. exit)
+For more information, please refer to our user guide at:https://ay2324s1-cs2113t-w11-1.github.io/tp/UserGuide.html
+____________________________________________________________
+```
````
A bit hard to read

junhyeong0411 commented:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/108390195/9cdf505c-325b-4619-941e-c9eca04ff962)

When you use exception on sequence diagram, I think using Alt block will be better.
Then you can also show that the activation bar brokes because of IllegalArgumentException.

junhyeong0411 commented:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/108390195/6b12e1d3-1a0d-4507-8e8a-48e011007a00)

In these parts, I think it is better to use alt blocks to show if-else statement.
Those bars are self-invocation bars, but actually your code doesn't self-invoke in that part.

junhyeong0411 commented:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/108390195/279c2a12-870e-4fd6-910c-440907dc570e)

When writing components name, I think matching the Uppercase and Lowercase with the actual code is better.
(For example, Ui, Parser, ...)

junhyeong0411 commented:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/108390195/68a63eb6-ba69-40f7-83b1-6bb291ef16e3)

In this part, I think user can be more detailed.

yingx9 committed:
Merge pull request #174 from JoanneJo/JoanneAng-DeveloperGuide

Update Developer Guide

JoanneJo committed:
Check Code Quality and Coding Standards

Created a Message.java class under UI to hold
messages that are meant to be output to the user.

Made minor update to DG.

JoanneJo committed:
Update EXPECTED.TXT

yingx9 committed:
Merge pull request #175 from JoanneJo/JoanneAng-CodeQualityStandards

Check Code Quality and Coding Standards

yingx9 committed:
Add Messages class specific to EditCommand and ListCommand

Update help message for edit and list

yingx9 committed:
Update EditCommand for minor change

yingx9 committed:
Fix checkstyle error

yingx9 committed:
Improve coding standard and quality

Remove unnecessary new lines and unused variables to improve
neatness and quality

yingx9 committed:
Update User Guide

yingx9 committed:
Fix user guide images

yingx9 committed:
Update DG

yingx9 committed:
Fix wrong argument message error for CD

000verflow committed:
Fix bugs in code and UG

yingx9 committed:
Update PPP and code quality

000verflow committed:
Improve code quality for Find Command

yingx9 committed:
Remove unused code

yingx9 committed:
Fix checkstyle error

000verflow committed:
Update Code quality for storage

yingx9 committed:
Merge pull request #178 from 000verflow/Update-misc

Update misc

yingx9 committed:
Merge remote-tracking branch 'upstream/master'
into YingXia-UpdateDevelopersGuide

# Conflicts:
#	src/main/java/seedu/ui/UI.java
#	src/test/java/seedu/commands/HelpCommandTest.java
#	src/test/java/seedu/parser/ParserTest.java
#	text-ui-test/EXPECTED.TXT

yingx9 committed:
Update help message

yingx9 committed:
Update editcommand help message

000verflow committed:
Merge pull request #176 from yingx9/YingXia-UpdateDevelopersGuide

Add Messages class specific to EditCommand and ListCommand

bnjm2000 committed:
Update UG

000verflow committed:
Update Storage with logging

bnjm2000 committed:
Fix checkstyle

bnjm2000 committed:
Update Expected.txt

000verflow committed:
Fix bug

bnjm2000 committed:
Update PPP

bnjm2000 committed:
Fix logger

000verflow committed:
Update DeveloperGuide.md

bnjm2000 committed:
Update checks

bnjm2000 committed:
Adds more logs

bnjm2000 committed:
Add assertions

000verflow committed:
Update DeveloperGuide.md

000verflow committed:
Update DG with content page

DavinciDelta committed:
Update DG

Update Command in DG with diagram

bnjm2000 committed:
change resourceList and eventList to plural form

yingx9 committed:
Add Component Diagram to Developer's Guide and update explanations

bnjm2000 committed:
Update tests

yingx9 committed:
Merge pull request #181 from DavinciDelta/UpdateGuides

Update DG

000verflow committed:
Update DG diagrams, and content

000verflow committed:
Update 000verflow.md

DavinciDelta committed:
Update DeveloperGuide.md

Update Non-Functional Requirements

DavinciDelta committed:
Merge pull request #184 from 000verflow/add-DGupdates

Add dg updates

DavinciDelta committed:
Merge pull request #182 from bnjm2000/BenjaminNg-FixLogsandConsistency

Fix logs, EventList and ResourceList

000verflow committed:
Merge branch 'master' into UpdateGuides

bnjm2000 committed:
Add EventEdit and Summary to DG

000verflow committed:
Merge pull request #185 from DavinciDelta/UpdateGuides

Update DeveloperGuide.md

000verflow committed:
Merge branch 'master' into YingXia-UpdateDGSections

bnjm2000 committed:
Add DG contributions

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-UpdatePPP

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Update DeveloperGuide.md
  Update 000verflow.md
  Update DG diagrams, and content
  Update tests
  change resourceList and eventList to plural form
  Update DG
  Update DG with content page
  Update DeveloperGuide.md
  Add assertions
  Adds more logs
  Update DeveloperGuide.md
  Fix logger
  Fix bug
  Update Storage with logging

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-Update-UG

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Update DeveloperGuide.md
  Update 000verflow.md
  Update DG diagrams, and content
  Update tests
  change resourceList and eventList to plural form
  Update DG
  Update DG with content page
  Update DeveloperGuide.md
  Add assertions
  Adds more logs
  Update DeveloperGuide.md
  Fix logger
  Fix bug
  Update Storage with logging

000verflow committed:
Merge pull request #183 from yingx9/YingXia-UpdateDGSections

Add Component Diagram to Developer's Guide and update explanations

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-Update-UG

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Add Component Diagram to Developer's Guide and update explanations

000verflow committed:
Update Diagram

000verflow committed:
Update DeveloperGuide.md

bnjm2000 committed:
Add Examples to Manual Testing

bnjm2000 committed:
Fix space before responds in ListCommand

yingx9 committed:
Merge pull request #186 from 000verflow/update-dg-diagram

Update dg diagram

bnjm2000 committed:
Add example outputs

bnjm2000 committed:
Add manual testing for Events

bnjm2000 committed:
Fix Conding Standard, Consistency and Quality

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-FixConsistency

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Update DeveloperGuide.md
  Update Diagram

bnjm2000 committed:
Fix tests

bnjm2000 committed:
Update Help Checks

bnjm2000 committed:
Beautify help message and shorten URL to UG

bnjm2000 committed:
Update test

bnjm2000 committed:
Update Test

bnjm2000 committed:
Fix Consistency

bnjm2000 committed:
Fix checkstyle

bnjm2000 committed:
Fix checkstyle

bnjm2000 committed:
Fix IO test
Bloody US server not caught up with SG time

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-UpdateUGandDG

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Update DeveloperGuide.md
  Update Diagram

JoanneJo committed:
Merge pull request #187 from bnjm2000/BenjaminNg-FixConsistency

Fix Consistency

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-UpdateUGandDG

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Fix IO test Bloody US server not caught up with SG time
  Fix checkstyle
  Fix checkstyle
  Fix Consistency
  Update Test
  Update test
  Beautify help message and shorten URL to UG
  Update Help Checks
  Fix tests
  Fix Conding Standard, Consistency and Quality
  Fix space before responds in ListCommand

# Conflicts:
#	src/main/java/seedu/ui/UI.java

bnjm2000 committed:
Update DG

bnjm2000 committed:
Update UG

bnjm2000 committed:
Update PPP

bnjm2000 committed:
Fix errata

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-UpdatePPP

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Fix IO test Bloody US server not caught up with SG time
  Fix checkstyle
  Fix checkstyle
  Fix Consistency
  Update Test
  Update test
  Beautify help message and shorten URL to UG
  Update Help Checks
  Fix tests
  Fix Conding Standard, Consistency and Quality
  Fix space before responds in ListCommand
  Update DeveloperGuide.md
  Update Diagram
  Add Component Diagram to Developer's Guide and update explanations

bnjm2000 committed:
Update PPP

bnjm2000 committed:
Remove unused import

bnjm2000 committed:
Add Tests to DeleteCommand and EventEditCommand

JoanneJo committed:
Merge branch 'master' into JoanneAng-ValidateAdd

yingx9 committed:
Merge pull request #180 from bnjm2000/BenjaminNg-UpdatePPP

Update PPP

yingx9 committed:
Merge pull request #188 from bnjm2000/BenjaminNg-UpdateUGandDG

Update UG & DG

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-AddTests

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Remove unused import
  Update PPP
  Fix errata
  Update PPP
  Update UG
  Update DG
  Add manual testing for Events
  Add example outputs
  Add Examples to Manual Testing
  Add DG contributions
  Add EventEdit and Summary to DG
  Update checks
  Update PPP
  Update Expected.txt
  Fix checkstyle
  Update UG

yingx9 committed:
Fix bugs in UG, update Target Users and PPP

yingx9 committed:
Merge remote-tracking branch 'upstream/master' into YingXia-UpdateDGSections

# Conflicts:
#	src/main/java/seedu/data/GenericList.java
#	src/test/java/seedu/commands/EditCommandTest.java

yingx9 committed:
Update files after merge

DavinciDelta committed:
Merge pull request #189 from yingx9/YingXia-UpdateDGSections

Fix bugs in UG, update Target Users and PPP

000verflow committed:
Update UG and PPP

yingx9 committed:
Fix DG formatting bug and update PPP

000verflow committed:
Add user centric language to UG

000verflow committed:
Update 000verflow.md

000verflow committed:
Update EXPECTED.TXT

yingx9 committed:
Merge pull request #191 from 000verflow/Update-UG-DG-PPP

Update ug dg ppp

yingx9 committed:
Update ListCommand to ignore capitalization

Users may enter keywords in lower case or upper case

Change ListCommand to find the correct resource regardless of
capitalization

yingx9 committed:
Update hasGenre code quality

yingx9 committed:
Merge remote-tracking branch 'upstream/master' into YingXia-UpdateListFeatureGenre

yingx9 committed:
Merge branch 'YingXia-UpdateListFeatureGenre' into YingXia-UpdateDGSections

yingx9 committed:
Update UG

bnjm2000 committed:
Add Tests

yingx9 committed:
Update example input in UG

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-AddTests

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Update EXPECTED.TXT
  Update 000verflow.md
  Add user centric language to UG
  Update UG and PPP
  Update files after merge
  Fix bugs in UG, update Target Users and PPP

# Conflicts:
#	src/main/java/seedu/data/GenericList.java
#	src/main/java/seedu/syslib/Syslib.java
#	src/test/java/seedu/commands/EditCommandTest.java
#	src/test/java/seedu/commands/ListCommandTest.java
#	src/test/java/seedu/util/TestUtil.java

yingx9 committed:
Update editcommand example inpu

000verflow committed:
Update DG UI component

bnjm2000 committed:
Merge pull request #190 from yingx9/YingXia-UpdateDGSections

Fix DG formatting bug and update PPP

bnjm2000 committed:
Fix checkstyle

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-AddTests

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Update editcommand example inpu
  Update example input in UG
  Update UG
  Update hasGenre code quality
  Update ListCommand to ignore capitalization
  Fix DG formatting bug and update PPP

bnjm2000 committed:
Update DG

bnjm2000 committed:
Fix lineseparator() error

DavinciDelta committed:
Update ppp

Update ppp name

000verflow committed:
Fix bug with storage not working on mac

000verflow committed:
Revert "Fix bug with storage not working on mac"

This reverts commit f21df6554960a3944cc3590617da1fea6f73f63f.

000verflow committed:
Revert "Update DG UI component"

This reverts commit 88f602941d423240b82a5ef504f6b10c827eff49.

bnjm2000 committed:
Update PPP

bnjm2000 committed:
Update DG

000verflow committed:
Update DG and Storage Bug

yingx9 committed:
Merge pull request #193 from DavinciDelta/master

Update ppp

yingx9 committed:
Merge pull request #194 from bnjm2000/BenjaminNg-UpdatePPPandDG

Update PPP and DG

000verflow committed:
Update find sequence diagram

000verflow committed:
Update DeveloperGuide.md

000verflow committed:
Fix storage file check, with universal paths library

DavinciDelta committed:
Update help command

Improve clarity of help

000verflow committed:
Bug fix in storage

000verflow committed:
Fix style

DavinciDelta committed:
Update help command test

000verflow committed:
Fix bug again

DavinciDelta committed:
Merge pull request #195 from 000verflow/update-dg-and-storage

Update dg and storage

yingx9 committed:
Merge pull request #196 from DavinciDelta/HelpUpdate

Update help command

yingx9 committed:
Merge pull request #192 from bnjm2000/BenjaminNg-AddTests

Add Tests, rename ResourceList and EventList

bnjm2000 committed:
Update sample input and output and formats for DG

bnjm2000 committed:
Add spacing for markdown on UG

bnjm2000 committed:
Update contributions on PPP

yingx9 committed:
Update display message

bnjm2000 committed:
Fix bar not appearing on Windows

DavinciDelta committed:
Improve feedback

Improve feedback from exceptions to provide more clarity

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-UpdatePPPandDG

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Fix bug again
  Update help command test
  Fix style
  Bug fix in storage
  Update help command
  Fix storage file check, with universal paths library
  Update DeveloperGuide.md
  Update find sequence diagram
  Update DG and Storage Bug
  Revert "Update DG UI component"
  Revert "Fix bug with storage not working on mac"
  Fix bug with storage not working on mac
  Update ppp
  Fix lineseparator() error
  Fix checkstyle
  Update DG UI component
  Add Tests
  Add Tests to DeleteCommand and EventEditCommand

# Conflicts:
#	src/test/java/seedu/commands/SummaryCommandTest.java

bnjm2000 committed:
Merge pull request #198 from yingx9/YingXia-UpdateArgsMessage

Update display message

DavinciDelta committed:
Update Command.java

improve variable name

bnjm2000 committed:
Removed unused import

yingx9 committed:
Merge pull request #200 from DavinciDelta/ArgumentHandling

Improve feedback

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-UpdatePPPandDG

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Update display message

# Conflicts:
#	src/main/java/seedu/ui/EditCommandMessages.java

yingx9 committed:
Merge pull request #199 from bnjm2000/BenjaminNg-UpdatePPPandDG

Update PPP, DG and Fix bar for Summary

JoanneJo committed:
Improve Validation for Add Command

Renamed ParseResource.java to ParseAttribute.java
for parsing related only to attributes.
Added a ParseResource.java to parse general input
for adding.

Validation added:
- Slash can only be used in 2 situations.
1. To indicate the type of information passed.
2. When it is wrapped with words on both sides.
- Multiple genres can be separated using comma.
- '[' and ']' are not allowed in genres.

JoanneJo committed:
Merge branch 'master' into JoanneAng-ValidateAdd

JoanneJo committed:
Fix Checkstyle

JoanneJo committed:
Update CreateResource.java

JoanneJo committed:
Update EXPECTED.TXT

JoanneJo committed:
Update EXPECTED.TXT

JoanneJo committed:
Merge branch 'JoanneAng-ValidateAdd' of https://github.com/JoanneJo/tp into JoanneAng-ValidateAdd

# Conflicts:
#	text-ui-test/EXPECTED.TXT

JoanneJo committed:
Merge branch 'master' into JoanneAng-ValidateAdd

JoanneJo committed:
Update EXPECTED.TXT

JoanneJo committed:
Update EXPECTED.TXT

bnjm2000 committed:
Merge pull request #201 from JoanneJo/JoanneAng-ValidateAdd

Improve Validation for Add Command

DavinciDelta committed:
Add CommandExceptionTesting

Including testing for exceptions in commands

yingx9 committed:
Merge pull request #202 from DavinciDelta/ArgumentHandling

Add CommandExceptionTesting

JoanneJo committed:
Merge branch 'master' into JoanneAng-ValidateAdd

yingx9 committed:
Update readme.md

DavinciDelta committed:
Update comments for suggestParser

DavinciDelta committed:
Merge pull request #203 from yingx9/YingXia-UpdateReadme.md

Update readme.md

yingx9 committed:
Update greeting message

yingx9 committed:
Merge pull request #204 from DavinciDelta/master

Update comments for suggestParser

yingx9 committed:
Merge remote-tracking branch 'upstream/master' into YingXia-UpdateReadme.md

JoanneJo committed:
Fix Invalid First Word Bug for Add Command

JoanneJo committed:
Update EXPECTED.TXT

JoanneJo committed:
Merge branch 'JoanneAng-ValidateAdd' into JoanneAng-DeveloperGuide

yingx9 committed:
Update List Screenshot

yingx9 committed:
Merge pull request #205 from JoanneJo/JoanneAng-ValidateAdd

Fix Invalid First Word Bug for Add Command

yingx9 committed:
Merge remote-tracking branch 'upstream/master' into YingXia-UpdateUG

yingx9 committed:
Fix formatting

DavinciDelta committed:
Update DG and UG

Include additional FAQ in UG

Include more information for parser in DG

DavinciDelta committed:
Merge branch 'master' of https://github.com/DavinciDelta/tp

DavinciDelta committed:
Merge pull request #206 from yingx9/YingXia-UpdateUG

Update UG Screenshot

DavinciDelta committed:
Update ppp

Update my ppp

DavinciDelta committed:
Merge branch 'AY2324S1-CS2113T-W11-1:master' into master

yingx9 committed:
Merge pull request #207 from DavinciDelta/master

Update DG and UG

DavinciDelta committed:
Update davincidelta.md

Update naming

000verflow committed:
Fix bug to work on mac os as well

000verflow committed:
Hardcode date to pass test

bnjm2000 committed:
Update UG

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-UpdatePPPandDG

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Update ppp
  Update DG and UG

bnjm2000 committed:
Merge pull request #210 from 000verflow/update-storage-filecreation

Fix bug to work on mac os as well

bnjm2000 committed:
Merge pull request #208 from DavinciDelta/master

Update davincidelta.md ppp

bnjm2000 committed:
Rearrange output of summaryCommand

bnjm2000 committed:
Update DG

bnjm2000 committed:
Update PPP

bnjm2000 committed:
Fix SummaryCommandTest

bnjm2000 committed:
Add checks and consistency

bnjm2000 committed:
Update EXPECTED.txt

bnjm2000 committed:
Add tests

bnjm2000 committed:
Fix EventDelete

bnjm2000 committed:
Add EditCommandTests

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-UpdatePPPandDG

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Hardcode date to pass test
  Fix bug to work on mac os as well
  Update davincidelta.md

bnjm2000 committed:
Clear logs

JoanneJo committed:
Update Developer Guide

bnjm2000 committed:
Update README with logo

JoanneJo committed:
Merge branch 'JoanneAng-DeveloperGuide' into JoanneAng-PPP

bnjm2000 committed:
Update DG

bnjm2000 committed:
Update PPP

bnjm2000 committed:
Merge pull request #214 from JoanneJo/JoanneAng-DeveloperGuide

Update Developer Guide

bnjm2000 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp into BenjaminNg-UpdatePPPandDG

* 'master' of https://github.com/AY2324S1-CS2113T-W11-1/tp:
  Update Developer Guide

bnjm2000 committed:
Update UG

bnjm2000 committed:
Remove "CLI" from exit message

bnjm2000 committed:
Update Tests

bnjm2000 committed:
Fix more tests

bnjm2000 committed:
Format UG with breaks

bnjm2000 committed:
Format DG with breaks for PDF

bnjm2000 committed:
Update main README with logo

JoanneJo committed:
Update Project Portfolio Page

JoanneJo committed:
Merge branch 'JoanneAng-PPP' into JoanneAng-AddCommandJUnit

JoanneJo committed:
Add More JUnit Tests

JoanneJo committed:
Fix Checkstyle

JoanneJo committed:
Merge pull request #211 from bnjm2000/BenjaminNg-UpdatePPPandDG

JoanneJo committed:
Merge pull request #213 from bnjm2000/BenjaminNg-FixEventDelete

DavinciDelta committed:
Update DeveloperGuide.md

Update use cases for events

DavinciDelta committed:
Update davincidelta.md

JoanneJo committed:
Merge branch 'master' into JoanneAng-PPP

DavinciDelta committed:
Update EXPECTED.TXT

fix bug for test

DavinciDelta committed:
Update EXPECTED.TXT

fix missed bugs

yingx9 committed:
Merge pull request #217 from DavinciDelta/master

Update DG

JoanneJo committed:
Merge remote-tracking branch 'upstream/master' into JoanneAng-PPP

# Conflicts:
#	docs/DeveloperGuide.md

000verflow committed:
Update DG usecase and UG

JoanneJo committed:
Merge branch 'JoanneAng-PPP' of https://github.com/JoanneJo/tp into JoanneAng-PPP

# Conflicts:
#	docs/DeveloperGuide.md

yingx9 committed:
Merge pull request #215 from JoanneJo/JoanneAng-PPP

Update Project Portfolio Page

000verflow committed:
Merge branch 'master' into Update-DG

JoanneJo committed:
Merge branch 'JoanneAng-PPP' into JoanneAng-AddCommandJUnit

# Conflicts:
#	src/test/java/seedu/commands/AddCommandTest.java

yingx9 committed:
Merge pull request #218 from 000verflow/Update-DG

Update DG usecase and UG

yingx9 committed:
Merge pull request #216 from JoanneJo/JoanneAng-AddCommandJUnit

Add More JUnit Tests

yingx9 committed:
Fix logo link

bnjm2000 committed:
Merge pull request #219 from yingx9/YingXia-FixLogoReadMe

yingx9 committed:
Fix logo formatting

yingx9 committed:
Merge remote-tracking branch 'upstream/master' into YingXia-FixLogoReadMe

yingx9 committed:
Fix images and formatting error

yingx9 committed:
Fix formatting error

yingx9 committed:
Update DG and UG formatting

JoanneJo committed:
Update Project Portfolio Page

bnjm2000 committed:
Prints Syslib version

bnjm2000 committed:
Merge pull request #220 from yingx9/YingXia-FixReadme

Fix ReadMe

JoanneJo committed:
Merge pull request #221 from JoanneJo/JoanneAng-PPP

Update Project Portfolio Page

bnjm2000 committed:
Update test

bnjm2000 committed:
Fix testp1

bnjm2000 committed:
Fix testp2

bnjm2000 committed:
fix test

yingx9 committed:
Merge pull request #222 from bnjm2000/BenjaminNg-addVersion

Add Syslib version printout

JoanneJo committed:
Merge remote-tracking branch 'upstream/master' into JoanneAng-UserGuide

JoanneJo committed:
Update UG

JoanneJo committed:
Merge pull request #223 from JoanneJo/JoanneAng-UserGuide

Update UG
